internal/engine/planner.go

```diff
@@ -424,7 +428,7 @@ func evaluateCELExprPartially(expr *exprpb.CheckedExpr, input *enginev1.Resource
 	val, details, err := prg.Eval(vars)
 	if err != nil {
 		if strings.HasPrefix(err.Error(), "no such key:") {
-			return nil, nil, fmt.Errorf("%w%s", ErrNoSuchKey, err)
+			return nil, nil, &NoSuchKeyError{msg: err.Error()}
```

Comment:
The error message is `{"code":3,"message":"Bad request: no such key: department"}` which is not very helpful. So let's make it a bit more friendlier.

```suggestion
			return nil, nil, &NoSuchKeyError{msg: fmt.Sprintf("missing principal attribute "%q", strings.TrimPrefix( err.Error(), "no such key: "))}
```